lib/internal/modules/esm/initialize_import_meta.js

```diff
@@ -60,9 +62,27 @@ function initializeImportMeta(meta, context, loader) {
   if (StringPrototypeStartsWith(url, 'file:') === true) {
     // These only make sense for locally loaded modules,
     // i.e. network modules are not supported.
-    const filePath = fileURLToPath(url);
-    meta.dirname = dirname(filePath);
-    meta.filename = filePath;
+
+    // Avoid paying the cost to derive these unless they're actually accessed.
+    ObjectDefineProperties(meta, {
+      dirname: {
```

Comment:
As written, I believe this would throw when attempting to set `import.meta.dirname`. The missing `configurable: true` is likely also the reason for the errors below. Can we have a test to show that `import.meta.dirname = 'foobar'` works in user code after this change?

Reply:
There wasn't, that I could find. So I added one :)